packages/test-base/src/androidTest/kotlin/io/realm/kotlin/test/shared/notifications/RealmListQueryTests.kt

```diff
@@ -0,0 +1,250 @@
+                    }
+            }
+            channel.receive().let { resultsChange ->
+                assertIs<ResultsChange<*>>(resultsChange)
+                assertNotNull(resultsChange.list)
+                assertEquals(6, resultsChange.list.size)
+            }
+            realm.writeBlocking {
+                val queriedSample = findLatest(querySample)
+                queriedSample!!.objectListField.add(QuerySample(intField = 1))
+            }
+            channel.receive().let { resultsChange ->
+                assertIs<ResultsChange<*>>(resultsChange)
+                assertNotNull(resultsChange.list)
+                assertEquals(7, resultsChange.list.size)
+            }
+            observer.cancel()
+            channel.close()
+        }
+    }
+
+    @Test
+    fun realmList_flowDeleteElement() {
+        runBlocking {
+            val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+            val channel = Channel<ResultsChange<*>>(capacity = 1)
+            val observer = async {
+                querySample.objectListField
+                    .query("TRUEPREDICATE").asFlow()
+                    .collect { flowList ->
+                        channel.send(flowList)
+                    }
+            }
+            channel.receive().let { resultsChange ->
+                assertIs<ResultsChange<*>>(resultsChange)
+                assertNotNull(resultsChange.list)
+                assertEquals(6, resultsChange.list.size)
+            }
+            realm.writeBlocking {
+                val queriedSample = findLatest(querySample)
+                queriedSample!!.objectListField.removeFirst()
+            }
+            channel.receive().let { resultsChange ->
+                assertIs<ResultsChange<*>>(resultsChange)
+                assertNotNull(resultsChange.list)
+                assertEquals(5, resultsChange.list.size)
+            }
+            observer.cancel()
+            channel.close()
+        }
+    }
+
+    @Test
+    fun realmList_flowDeleteList() {
+        runBlocking {
+            val querySample = realm.query<QuerySample>("intField == $0", 1).find().first()
+            val channel = Channel<ResultsChange<*>>(capacity = 1)
+            val observer = async {
+                querySample.objectListField
+                    .query("TRUEPREDICATE").asFlow()
+                    .collect { flowList ->
+                        channel.send(flowList)
+                    }
+            }
+            channel.receive().let { resultsChange ->
+                assertIs<ResultsChange<*>>(resultsChange)
+                assertNotNull(resultsChange.list)
+                assertEquals(6, resultsChange.list.size)
+            }
+            realm.writeBlocking {
+                val queriedSample = findLatest(querySample)
+                queriedSample!!.objectListField = realmListOf()
+            }
+            channel.receive().let { resultsChange ->
```

Comment:
Does this test actually parse. I would have expected the flow to complete without sending an event. I.e. you need something like:

```
                    .query("TRUEPREDICATE").asFlow()
                    .onCompletion { channel.send(/*something*/) }
                    .collect { flowList ->
                        channel.send(flowList)
                    }
```
```